pyresample/test/test_gradient.py

```diff
@@ -624,3 +628,23 @@ def test_index_search_with_data_requested_outside_top_left_boundary(self):
                                                  self.dst_x, self.dst_y)
         np.testing.assert_allclose(res_x, expected_x)
         np.testing.assert_allclose(res_y, expected_y)
+
+
+def test_resampling_geos_edge_to_mercator():
+    """Test that projecting the edges of geos onto a mercator area does not produce unnecessary NaNs."""
+    source_area = AreaDefinition.from_extent(area_id="side",
+                                             projection={'ellps': 'WGS84', 'h': '35786400', 'lon_0': '0',
+                                                         'no_defs': 'None', 'proj': 'geos', 'type': 'crs',
+                                                         'units': 'm', 'x_0': '0', 'y_0': '0'},
+                                             area_extent=(-2483999.9974, 5121999.9947, -1739999.9982, 4809999.995),
+                                             shape=(156, 372))
+    dest_area = AreaDefinition.from_extent(area_id="dest",
+                                           projection={'a': '6378137', 'b': '6378137', 'k': '1', 'lat_ts': '0',
+                                                       'lon_0': '0', 'nadgrids': '@null', 'no_defs': 'None',
+                                                       'proj': 'merc', 'type': 'crs', 'units': 'm', 'wktext': 'None',
+                                                       'x_0': '0', 'y_0': '0'},
+                                           area_extent=(-7800000.0, 8595618.56, -6073037.8378, 10321713.92),
+                                           shape=(512, 512))
+
+    res = gradient_resampler_indices(source_area, dest_area, fill_value=np.nan)
+    assert not np.any(np.isnan(res[:, :, -1]))
```

Comment:
What are the dimensions of `res`? Why is it 3D?

Reply:
the first dimension is of length 2, containing the x and y components of the indices. The others are lines and columns.